nilearn/surface/surface.py

```diff
@@ -460,6 +460,60 @@ def _projection_matrix(
     return proj
 
 
+def _mask_sample_locations(sample_locations, img_shape, mesh_n_vertices, mask):
+    """Mask sample locations without changing to indices."""
+    sample_locations = np.asarray(np.round(sample_locations), dtype=int)
+    masks = _masked_indices(np.vstack(sample_locations), img_shape, mask=mask)
+    masks = np.split(masks, mesh_n_vertices)
+    # mask sample locations and make a list of masked indices because
+    # masked locations are not contiguous
+    masked_sample_locations = [
+        sample_locations[idx][~mask] for idx, mask in enumerate(masks)
+    ]
+    return masked_sample_locations
+
+
+def _nearest_most_frequent(
```

Comment:
So now I have separated the nearest most frequent as a new interpolation method.

I also noticed that, right now, we only use vol_to_surf to project stat maps in our examples.

So it would be nice to have one where we demonstrate this for atlases (and also compare the different methods for different use cases). In a separate PR though. LMWYT @Remi-Gau @bthirion